src/main/java/org/jabref/model/entry/identifier/DOI.java

```diff
@@ -141,6 +145,47 @@ public DOI(String doi) {
         }
     }
 
+    /**
+     * Removes all special characters at the end of the DOI string
+     *
+     * @param doiStr the DOI/Short DOI string
+     * @return an DOI string with the special character at the end removed
+     */
+    public static String removeScharDOI(String doiStr) {
+        // jie add
+        char[] lcharDoi = doiStr.toCharArray();
+        int i = lcharDoi.length - 1;
+        // valid DOI characters are a-z(97-122), A-Z(65-90), 0-9(48-57), -(45), .(46), _(95), ;(59), ( (40), ) (41), /(47)
+        for (; i >= 0; i--) {
+            if ((lcharDoi[i] >= 45 && lcharDoi[i] <= 47) ||
+                    (lcharDoi[i] >= 48 && lcharDoi[i] <= 57) ||
+                    (lcharDoi[i] >= 65 && lcharDoi[i] <= 90) ||
+                    (lcharDoi[i] >= 97 && lcharDoi[i] <= 122) ||
+                    lcharDoi[i] == 95 ||
+                    lcharDoi[i] == 59 ||
+                    lcharDoi[i] == 40 ||
+                    lcharDoi[i] == 41
+            ) {
+                // System.out.println(i);
+                break;
+            }
+        }
+
+//        char[] lcharDoi = doiStr.toCharArray();
```

Comment:
remove commented out code

Reply:
This is not really in the spirit of open source contribution. You requested to work on this issue after me and I have started working on it.

Reply:
> This is not really in the spirit of open source contribution. You requested to work on this issue after me and I have started working on it.

I actually started working on this issue 1 week ago. Then I saw your request, I thought maybe I should also post my progress. Do we have to first get permission before we can work on the open issues? 

thanks,

Reply:
> remove commented out code

thanks very much for the suggestions, will do.

Reply:
> and I have started working on it.

@mrcstan May we see your approach? Maybe it differs significantly of this one - and we can make use of your ideas, too?